account_payment_order/models/account_payment_order.py

```diff
@@ -331,11 +334,11 @@ def draft2open(self):
                 with self.env.norecompute():
                     payline.date = requested_date
                 # Group options
-                if order.payment_mode_id.group_lines:
-                    hashcode = payline.payment_line_hashcode()
-                else:
-                    # Use line ID as hascode, which actually means no grouping
-                    hashcode = payline.id
+                hashcode = (
```

Comment:
Now that the diff is clearer, I'm seeing that previous code was OK. Why did you initially change it? I think it's better to leave it as it was for reducing diff. Sorry for changing my mind.

Reply:
I know but the pre-commit didn't let me commit as it was because of complexity factor that was too high. This is the reason why i modified this part

Reply:
OK then.